### PR TITLE
Chapter 3

### PR DESCRIPTION
**Pull request by umairah97**

### CODE REVIEW

**Comment by umairah97** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Your PR is having 14 commits. You must merge to 1 commit, then ready again

**Comment by umairah97** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Please. Do same as below command
```
git checkout master
git pull origin master
git checkout chapter_3
git rebase master
->>> Fix conflict
git add .
git rebase --continue
git push origin -f chapter_3
```

**Comment by umairah97** [comment]
ready

**Comment by manhnt-0960** [comment]
commented
please merge 2 commits to 1

**Comment by umairah97** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,11 @@ source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "2.7.0"
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline Gemfile:5]: resolve conflict, please

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/static_pages.scss:3]: remove file that never used

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb:6]:
```suggestion
  def home; end

  def help; end
```

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+        end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb:5]: indenting 2 spaces

File: app/helpers/static_pages_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
+end
```
Comment (saobv-2223) [inline app/helpers/static_pages_helper.rb:2]: remove file that never used

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+   <%= link_to "sample app", "#", id: "logo" %>
+   <nav>
+     <ul class ="nav navbar-nav navbar-right">
+        <li><%= link_to "Home", "#" %></li>
+        <li><%= link_to "Help" , "#" %></li>
+         <li><%= link_to "Log in" , "#" %></li>
+        </ul>
+    </nav>
+  </div>
+</header>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb:12]: add an empty line below line 12

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title (yield (:title)) %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb:4]: remove ()

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -10,6 +10,11 @@
   </head>
 
   <body>
-    <%= yield %>
-  </body>
+  <%= render "layouts/header " %>
+  <div class
+ ="container">
+  <%= yield %>
+  </div>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb:17]:
```suggestion
  <%= render "layouts/header " %>
  <div class="container">
    <%= yield %>
  </div>
```

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<%= provide : title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb:6]: using link_to instead of a tag

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<%= provide : title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a href="http://www.railstutorial.org/">Ruby on
+Rails Tutorial</a>
+sample application.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb:9]: add an empty line below line 9

File: config/routes.rb [hunk 11]
```diff
@@ -1,2 +1,4 @@
 Rails.application.routes.draw do
+  root 'static_pages#home'
+  get 'static_pages/help'
```
Comment (saobv-2223) [inline config/routes.rb:3]: using "

**Comment by umairah97** [comment]
ready

File: app/assets/javascripts/static_pages.coffee [hunk 12]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (saobv-2223) [inline app/assets/javascripts/static_pages.coffee:3]: remove file that never used

File: app/assets/stylesheets/custom.scss [hunk 13]
```diff
@@ -0,0 +1,33 @@
+
+/* universal */
+body {
+ padding-top: 60px;
+}
+section {
+ overflow: auto;
+}
+textarea {
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss:9]: each block should have an empty line among them

File: app/assets/stylesheets/custom.scss [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+ color: #fff;
+ text-transform: uppercase;
+ letter-spacing: -1px;
+ padding-top: 9px;
+ font-weight: bold;
+}
+#logo:hover {
+ color: #fff;
+ text-decoration: none;
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss:33]: add an emtpy line below line 33

File: app/assets/stylesheets/static_pages.scss (hunk 2 above)
Comment (saobv-2223) [reply app/assets/stylesheets/static_pages.scss:3]: fix it, please

File: app/helpers/application_helper.rb [hunk 15]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = "  "
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+        end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb:5]:
```suggestion
  def full_title page_title
    base_title = "Ruby on Rails Tutorial Sample App"
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
  end
```

File: app/helpers/static_pages_helper.rb (hunk 5 above)
Comment (saobv-2223) [inline app/helpers/static_pages_helper.rb:2]: delete this file

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -0,0 +1,12 @@
+<header class ="navbar navbar-fixed-top navbar-inverse">
+  <div class ="container">
+   <%= link_to "sample app", "#", id: "logo" %>
+   <nav>
+     <ul class ="nav navbar-nav navbar-right">
+        <li><%= link_to "Home", "#" %></li>
+        <li><%= link_to "Help" , "#" %></li>
+         <li><%= link_to "Log in" , "#" %></li>
+        </ul>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb:9]:
```suggestion
      </ul>
```

File: app/views/static_pages/help.html.erb [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<%= link_to provide : title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb:6]:
```suggestion
<%= link_to "Rails Tutorial", "http://www.railstutorial.org/help" %>
```

File: app/views/static_pages/help.html.erb [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+help page</a>.
+To get help on this sample app, see the
+<a href="http://www.railstutorial.org/book"><em>Ruby on Rails
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb:9]: same

File: app/views/static_pages/home.html.erb [hunk 19]
```diff
@@ -0,0 +1,9 @@
+<%= provide : title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a href="http://www.railstutorial.org/">Ruby on
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb:6]: same

**Comment by umairah97** [comment]
ready

File: Gemfile [hunk 20]
```diff
@@ -12,7 +12,13 @@ gem "coffee-rails", "~> 4.2"
 gem "turbolinks", "~> 5"
 gem "jbuilder", "~> 2.5"
 gem "bootsnap", ">= 1.1.0", require: false
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline Gemfile:15]: resolve conflict, please

File: app/views/static_pages/help.html.erb [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<%= link_to "Rails Tutorial", "http://www.railstutorial.org/help" %>
+help page</a>.
+To get help on this sample app, see the
+<%= link_to "Ruby on Rails", "http://www.railstutorial.org/book" %>
+Tutorial</em>
+book</a>.
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb:11]:
```suggestion
  Get help on the Ruby on Rails Tutorial at the
  <%= link_to "Rails Tutorial", "http://www.railstutorial.org/help" %>
  help page
  To get help on this sample app, see the
  <%= link_to "Ruby on Rails", "http://www.railstutorial.org/book" %>
  Tutorial</em>
  book
```

File: app/views/static_pages/home.html.erb [hunk 22]
```diff
@@ -0,0 +1,8 @@
+<%= link_to provide : title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<%= link_to "Ruby on Rails Tutorial", "http://www.railstutorial.org/" %>
+sample application.
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb:7]:
```suggestion
  This is the home page for the
  <%= link_to "Ruby on Rails Tutorial", "http://www.railstutorial.org/" %>
  sample application.
```

**Comment by umairah97** [comment]
ready

File: .gitignore [hunk 23]
```diff
@@ -60,3 +60,5 @@ build-iPhoneSimulator/
 config/credentials.yml.enc.example
 config/database.yml
 test
+app/assets/stylesheets/static_pages.scss
+app/helpers/static_pages_helper.rb
```
Comment (chungpv-1008) [inline .gitignore:64]: You should delete 2 that files and not move they to '.gitignore' file

File: app/assets/stylesheets/custom.scss [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+body
+
+{
+ padding-top: 60px;
+}
+section 
+
+{
+ overflow: auto;
+}
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss:12]:
```suggestion
body {
  padding-top: 60px;
}

section {
  overflow: auto;
}

```

File: app/assets/stylesheets/custom.scss [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+
+{
+ text-align: center;
+}
+
+.center h1
+
+{
+ margin-bottom: 10px;
+}
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss:29]:
```suggestion
textarea {
  resize: vertical;
}

.center {
  text-align: center;
}

.center h1 {
  margin-bottom: 10px;
}

```

File: app/assets/stylesheets/custom.scss [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+ padding-top: 9px;
+ font-weight: bold;
+ }
+
+#logo:hover 
+
+{
+ color: #fff;
+ text-decoration: none;
+}
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss:50]:
```suggestion
/* header */
#logo {
  float: left;
  margin-right: 10px;
  font-size: 1.7em;
  color: white;
  text-transform: uppercase;
  letter-spacing: -1px;
  padding-top: 9px;
  font-weight: bold;
 }

#logo:hover {
  color: white;
  text-decoration: none;
}

```

**Comment by umairah97** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**